tests/unit/components/sectioning/sectioningConfirmation.spec.tsx

```diff
@@ -0,0 +1,219 @@
+import { describe } from '@jest/globals';
+import '@testing-library/jest-dom';
+import { cleanup, render, screen, waitFor } from '@testing-library/react';
+import React from 'react';
+import Confirm from '../../../../src/pages/sectioning/Confirm';
+import { uniqueId } from 'lodash';
+import { enableMapSet } from 'immer';
+import labwareFactory from '../../../../src/lib/factories/labwareFactory';
+import { createMemoryRouter, RouterProvider } from 'react-router-dom';
+import * as reactDom from 'react-router-dom';
+import userEvent from '@testing-library/user-event';
+
+afterEach(() => {
+  cleanup();
+  jest.clearAllMocks();
+});
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useLocation: jest.fn(),
+  useLoaderData: jest.fn().mockReturnValue({
+    comments: [],
+    slotRegions: []
+  }),
+  useNavigate: jest.fn().mockImplementation(() => ({
+    navigate: jest.fn()
+  }))
+}));
+
+describe('Render Confirm Component', () => {
+  beforeEach(() => {
+    enableMapSet();
+  });
+  describe('when no state is specified', () => {
+    beforeEach(() => {
+      jest.spyOn(reactDom, 'useLocation').mockImplementation(() => {
+        return {
+          key: uniqueId(),
+          pathname: '/sectioning/confirm',
+          search: '',
+          hash: '',
+          state: null
+        };
+      });
+      renderConfirm();
+    });
+    it('loads all the page fields correctly', () => {
+      expectInitSectioningConfirmPage();
+    });
+    describe('After confirming some labware(s)', () => {
+      beforeEach(() => {
+        jest.mock('../../../../src/components/sectioningConfirm/SectioningConfirm', () => {
+          return {
+            __esModule: true,
+            default: jest.fn(({ onConfirmed }) => {
+              return <button onClick={() => onConfirmed([labwareFactory.build({ id: 1 })])}>Mock Confirm</button>;
+            })
+          };
+        });
+      });
+
+      it('reloads to Sectioning - Plan page when Reset Form button is clicked', () => {
+        waitFor(async () => {
+          await userEvent.click(screen.getByRole('button', { name: 'Reset Form' }));
+          expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('Sectioning - Plan');
+          expect(screen.getByRole('button', { name: 'Next' })).toBeDisabled();
```

Comment:
While looking for the slotMeasurement issue, I saw an article recommending the following way of testing results of an action as as the waitFor function deals with the situation  when you we to wait for an element to appear or change, or when you expect some action to occur after a certain period of time.

Do you think this is a better way?

For e.g
```
 it('reloads to Sectioning - Plan page when Reset Form button is clicked', async () => {
          await userEvent.click(screen.getByRole('button', { name: 'Reset Form' }));
          await waitFor(() => {
              expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('Sectioning - Plan');
              expect(screen.getByRole('button', { name: 'Next' })).toBeDisabled();
         })
})
```

Reply:
@seenanair technically that is the correct way to use it, but I found some tests fails if the action and the checks are not in the same await block.